application/models/user.go

```diff
@@ -556,3 +556,70 @@ func (u *User) GetPreference(key string) (*UserPreference, error) {
 
 	return &uPref, nil
 }
+
+func (u *User) CreatePreference(key, value string) (UserPreference, error) {
+	uPref := UserPreference{}
+
+	if err := u.FindByUUID(u.Uuid.String(), "id"); err != nil {
+		err := fmt.Errorf("can't create UserPreference - error finding user with uuid %s ... %v",
+			u.Uuid.String(), err)
+		return UserPreference{}, err
+	}
+
+	DB.Where("user_id = ?", u.ID).Where("key = ?", key).First(&uPref)
+	if uPref.ID > 0 {
+		err := fmt.Errorf("can't create UserPreference with key %s.  Already exists with id %v.", key, uPref.ID)
+		return UserPreference{}, err
+	}
+
+	uPref.Uuid = domain.GetUuid()
```

Comment:
Perhaps the `Save` function in `UserPreference` can get the UUID. I don't think we've done this consistently, but I have started to think it might be a good idea.

Reply:
Sounds good.